lms/djangoapps/courseware/views/views.py

```diff
@@ -497,10 +420,52 @@ def get(self, request, course_id, tab_type, **kwargs):
         course_key = CourseKey.from_string(course_id)
         with modulestore().bulk_operations(course_key):
             course = get_course_with_access(request.user, 'load', course_key)
-            tab = CourseTabList.get_tab_by_type(course.tabs, tab_type)
-            page_context = self.create_page_context(request, course=course, tab=tab, **kwargs)
-            set_custom_metrics_for_course_key(course_key)
-            return super(CourseTabView, self).get(request, course=course, page_context=page_context, **kwargs)
+            try:
+
+                # Verify that the user has access to the course
+                redirect_if_needed_to_access_course(request, course)
```

Comment:
I don't think we can handle course-specific redirects in middleware because the context isn't available (easily?). My thinking is that we want to convert all the tabs to render as fragments and use CourseTabView so that they benefit from the shared logic. I'm not sure I want to sprinkle TODOs around the code for that. I suppose I could make a ticket for each tab, but I don't know that that will ever get prioritized into a sprint.